test/framework/src/main/java/org/elasticsearch/logsdb/datageneration/datasource/DefaultPrimitiveTypesHandler.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.logsdb.datageneration.datasource;
+
+import org.elasticsearch.test.ESTestCase;
+
+import java.math.BigInteger;
+
+public class DefaultPrimitiveTypesHandler implements DataSourceHandler {
+    public DataSourceResponse handle(DataSourceRequest request) {
```

Comment:
Similarly, this looks like a good fit for inheritance?

Reply:
Sorry i don't fully see what's the inheritance structure.

Reply:
I think i figured it out.